### PR TITLE
Add Google Cloud DNS auth scope to GCE VMs when they are started in a federation.

### PR DESCRIPTION
**Pull request by madhusudancs**
cc @colhom @kubernetes/sig-cluster-federation 

[Analytics]()

### CODE REVIEW

**Comment by nikhiljindal** [comment]
LGTM, thanks!

I assume you will do a similar change for GKE and AWS as well. Doing it in a separate PR is fine.

**Comment by nikhiljindal** [comment]
Adding P1 since this should help in fixing federation e2e tests

**Comment by madhusudancs** [comment]
@nikhiljindal I do plan to add this to GKE scripts. I don't have a plan for AWS yet because I don't know exactly how the AWS API auth works. I probably need to add the IAM policies to the policies in `cluster/aws/templates/iam/` but I am not that well versed with AWS stuff. If someone who has a better understanding of AWS APIs want to give it a shot, I will be happy to let them do that.

**Comment by nikhiljindal** [comment]
Maybe file an issue and cc Justin?
We will need it in 1.3 for federation to work on AWS right?

File: cluster/gce/config-default.sh [hunk 1]
```diff
@@ -56,7 +56,12 @@ MASTER_TAG="${INSTANCE_PREFIX}-master"
 NODE_TAG="${INSTANCE_PREFIX}-minion"
 MASTER_IP_RANGE="${MASTER_IP_RANGE:-10.246.0.0/24}"
 CLUSTER_IP_RANGE="${CLUSTER_IP_RANGE:-10.244.0.0/14}"
-NODE_SCOPES="${NODE_SCOPES:-compute-rw,monitoring,logging-write,storage-ro}"
+if [[ "${FEDERATION:-}" == true ]]; then
+    NODE_SCOPES="${NODE_SCOPES:-compute-rw,monitoring,logging-write,storage-ro,https://www.googleapis.com/auth/ndev.clouddns.readwrite}"
```
Comment (quinton-hoole) [inline cluster/gce/config-default.sh]: I'm curious why the format of the clouddns scope is inconsistent with all the others. Is this necessary?  Can it be expressed in a manner consistent with the others?

Comment (quinton-hoole) [inline cluster/gce/config-default.sh]: Never mind.  I checked the docs, and it seems that there is no short-form alias for cloud DNS, so what you have is the best we can do.

**Comment by quinton-hoole** [comment]
@madhusudancs @nikhiljindal @justinsb It looks as if AWS does not require NODE_SCOPES:

https://github.com/kubernetes/kubernetes/blob/master/cluster/aws/config-default.sh#L86

@justinsb can you confirm?

Also @madhusudancs it looks like you need a similar addition here, for gce/config-test:

https://github.com/kubernetes/kubernetes/blob/447ce334ad3993398f95ad0a7519ab83845c918f/cluster/gce/config-test.sh#L60

Q

**Comment by madhusudancs** [comment]
> @madhusudancs @nikhiljindal @justinsb It looks as if AWS does not require NODE_SCOPES:

> https://github.com/kubernetes/kubernetes/blob/master/cluster/aws/config-default.sh#L86

Please see my comment here - https://github.com/kubernetes/kubernetes/pull/27357#issuecomment-225983853. We will probably need an IAM policy for Route53.

**Pull request merged**